### PR TITLE
Avoid unnecessary `cloneDeep` usage

### PR DESCRIPTION
**Pull request by Page-**
Change-type: patch

Resolves: # 
HQ: <url> 
See: <url> 
Depends-on: <url> 
Change-type: major|minor|patch 

---
##### Contributor checklist

- [ ] Includes tests
- [ ] Includes typings
- [ ] Includes updated documentation
- [ ] Includes updated build output

### CODE REVIEW

File: lib/util/index.ts [hunk 1]
```diff
@@ -87,7 +86,7 @@ export const mergePineOptions = <R extends {}>(
 		return defaults;
 	}
 
-	const result = cloneDeep(defaults);
+	const result = { ...defaults };
```
Comment (Page-) [inline lib/util/index.ts:89]: We clone the top level object as we mutate it heavily, for the lower levels only $expand actually gets mutated and we handle that separately

File: lib/util/index.ts [hunk 2]
```diff
@@ -145,7 +144,8 @@ const mergeExpandOptions = <T>(
 		return extraExpand;
 	}
 
-	defaultExpand = convertExpandToObject(defaultExpand);
+	// We only need to clone the defaultExpand as it's the only one we mutate
+	defaultExpand = convertExpandToObject(defaultExpand, true);
```
Comment (Page-) [inline lib/util/index.ts:148]: As the comment says ^

File: lib/util/index.ts [hunk 3]
```diff
@@ -221,5 +222,9 @@ const convertExpandToObject = <T extends {}>(
 		}
 	}
 
-	return cloneDeep(expandOption);
+	if (cloneIfNeeded) {
+		return { ...(expandOption as Pine.ResourceExpandFor<T>) };
+	}
```
Comment (Page-) [inline lib/util/index.ts:227]: We only need to clone here since the other steps return a object

**Comment by thgreasi** [comment]
Really nice :+1:

**Pull request merged**